Remove deprecated `--no-update` flag from Poetry lock command in CI workflow

After updating Poetry to version 2.0, the CI workflow is failing with the following error:

```
Run poetry lock --no-update
The option "--no-update" does not exist
Error: Process completed with exit code 1.
```

This is because the `--no-update` flag has been removed in Poetry 2.0, as its behavior is now the default. The fix requires removing the `--no-update` flag from the `poetry lock` command in the workflow file.

**Steps to Reproduce:**
1. Trigger the CI workflow on any pull request.
2. Observe the error in the `Poetry lock no-update` step.

**Expected Behavior:**  
The CI workflow should complete successfully without errors related to Poetry.

**Proposed Fix:**  
- Remove `--no-update` from the `poetry lock` command in the workflow file.
- Verify that the CI workflow passes after the change.

**References:**  
- [Poetry 2.0 Release Notes](https://python-poetry.org/blog/announcing-poetry-2.0.0/)

--- a/.github/workflows/test-package.yml
+++ b/.github/workflows/test-package.yml
@@ -33,8 +33,8 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
         cache: "poetry"
-    - name: Poetry lock no-update
-      run: poetry lock --no-update
+    - name: Poetry lock
+      run: poetry lock
     - name: Install dependencies
       run: poetry install --with=dev
     - name: Run unit tests

